internal/controller/services/servicemesh/servicemesh_controller_actions.go

```diff
@@ -0,0 +1,398 @@
+	}
+
+	return nil
+}
+
+func updateAuthRefsConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	sm, ok := rr.Instance.(*serviceApi.ServiceMesh)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a serviceApi.ServiceMesh)", rr.Instance)
+	}
+
+	audiences := sm.Spec.Auth.Audiences
+	audiencesList := ""
+	if audiences != nil && len(*audiences) > 0 {
+		audiencesList = strings.Join(*audiences, ",")
+	}
+
+	authorinoNamespace, err := getAuthorinoNamespace(rr)
+	if err != nil {
+		return fmt.Errorf("failed to obtain Authorino namespace from ServiceMesh CR: %w", err)
+	}
+
+	data := map[string]string{
+		"AUTH_AUDIENCE":   audiencesList,
+		"AUTH_PROVIDER":   authProviderName,
+		"AUTH_NAMESPACE":  authorinoNamespace,
+		"AUTHORINO_LABEL": authorinoLabel,
+	}
+
+	authRefsConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      authRefsConfigMapName,
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+		},
+		Data: data,
+	}
+	if err := controllerutil.SetControllerReference(sm, authRefsConfigMap, rr.Client.Scheme()); err != nil {
+		return fmt.Errorf("error setting owner reference to ConfigMap: %s", authRefsConfigMapName)
+	}
+
+	if err := rr.AddResources(authRefsConfigMap); err != nil {
+		return fmt.Errorf("error adding resource (ConfigMap): %s", authRefsConfigMapName)
+	}
+
+	return nil
+}
+
+func deleteFeatureTrackers(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ftNames := []string{
+		rr.DSCI.Spec.ApplicationsNamespace + "-mesh-shared-configmap",
+		rr.DSCI.Spec.ApplicationsNamespace + "-mesh-control-plane-creation",
+		rr.DSCI.Spec.ApplicationsNamespace + "-mesh-metrics-collection",
+		rr.DSCI.Spec.ApplicationsNamespace + "-enable-proxy-injection-in-authorino-deployment",
+		rr.DSCI.Spec.ApplicationsNamespace + "-mesh-control-plane-external-authz",
+	}
+
+	for _, n := range ftNames {
+		ft := featuresv1.FeatureTracker{}
+		err := rr.Client.Get(ctx, client.ObjectKey{Name: n}, &ft)
+		if k8serr.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return fmt.Errorf("failed to lookup FeatureTracker %s: %w", ft.GetName(), err)
+		}
+
+		err = rr.Client.Delete(ctx, &ft, client.PropagationPolicy(metav1.DeletePropagationForeground))
+		if k8serr.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return fmt.Errorf("failed to delete FeatureTracker %s: %w", ft.GetName(), err)
+		}
+	}
+
+	return nil
+}
+
+func checkSMCPReadiness(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	sm, ok := rr.Instance.(*serviceApi.ServiceMesh)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a serviceApi.ServiceMesh)", rr.Instance)
+	}
+
+	smcp := &unstructured.Unstructured{}
+	smcp.SetGroupVersionKind(gvk.ServiceMeshControlPlane)
+	err := rr.Client.Get(ctx, client.ObjectKey{
+		Name:      sm.Spec.ControlPlane.Name,
+		Namespace: sm.Spec.ControlPlane.Namespace,
+	}, smcp)
+
+	if err != nil {
```

Comment:
I'd explicitly check for not found, and return the error for other errors, so we won't swallow technical errors